Source/Core/Updater/Main.cpp

```diff
@@ -469,13 +473,22 @@ Manifest::Hash ComputeHash(const std::string& contents)
   return out;
 }
 
+int ProgressCallback(void*, double total, double now, double, double)
+{
+  UI::SetProgress(static_cast<int>(now), static_cast<int>(total));
+  return 0;
+}
+
 bool DownloadContent(const std::vector<Manifest::Hash>& to_download,
                      const std::string& content_base_url, const std::string& temp_path)
 {
-  Common::HttpRequest req(std::chrono::seconds(30));
+  Common::HttpRequest req(std::chrono::seconds(30), ProgressCallback);
+
   for (const auto& h : to_download)
   {
     std::string hash_filename = HexEncode(h.data(), h.size());
+    UI::SetDescription("Downloading " + hash_filename + "...");
```

Comment:
Why not use the actual filename here? Needs a bit of plumbing but much more user friendly and really not so hard :)